placeos

```diff
@@ -0,0 +1,303 @@
+echo '░░░░░░░  |░░ | \____░░\ ░░  _____|░░  __░░\ ░░ |  ░░ |\░░░░░░\'
+echo '░░  ____/ ░░ | ░░░░░░░ |░░ /      ░░░░░░░░ |░░ |  ░░ | \____░░\'
+echo '░░ |      ░░ |░░  __░░ |░░ |      ░░   ____|░░ |  ░░ |░░\   ░░ |'
+echo '░░ |      ░░ |\░░░░░░░ |\░░░░░░░\ \░░░░░░░\  ░░░░░░  |\░░░░░░  |'
+echo '\__|      \__| \_______| \_______| \_______| \______/  \______/'
+echo
+}
+
+abort() {
+    printf "░░░ ${red}${1}${reset}\n"
+    echo "░░░ Logs can be found in ${logfile}"
+    exit 1
+}
+
+run_or_abort() {
+    task="${1}"
+    start="${2}"
+    failure="${3}"
+
+    exit_code="0"
+    start_spinner "░░░ ${start}"
+    $(${task} >>${logfile} 2>&1) || exit_code="$?"
+    stop_spinner $exit_code
+
+    [ $exit_code -eq "0" ] || abort "${failure}"
+}
+
+hard_reset() {
+    run_or_abort \
+        "${base_path}/scripts/run-sam-task drop:db" \
+        "Dropping RethinkDB tables..." \
+        "Failed to drop RethinkDB tables."
+
+    run_or_abort \
+        "${base_path}/scripts/run-sam-task drop:elastic" \
+        "Dropping Elasticsearch indices..." \
+        "Failed to drop Elasticsearch indices."
+
+    run_or_abort \
+        'COMPOSE_PROJECT_NAME=placeos docker-compose restart nginx' \
+        "Restarting nginx..." \
+        "Failed to restart nginx."
+}
+
+start_environment__usage() {
+    cat <<EOF
+Usage: ./placeos start [-h|--help]
+
+Start the PlaceOS environment
+
+Arguments:
+    --hard-reset            Reset the environment to a default state.
+    -s, --sentry            Set-up Sentry
+    -h, --help              Display this message
+EOF
+}
+
+start_environment() {
+    SERVICES=('')
+    hard_reset=false
+    setup_sentry=false
+    while [ ${#} -gt 0 ]; do
+        command="${1}"
+        shift
+        case ${command} in
+        --hard-reset)
+            read -p "${red}Warning:${reset} This will reset your environment. Would you like to continue (y/n)?" choice
+            case "${choice}" in
+            y | Y | YES | yes)
+                hard_reset=true
+                ;;
+            *)
+                echo "${red}exiting.${reset}"
+                exit 1
+                ;;
+            esac
+            ;;
+        -s | --sentry)
+            setup_sentry=true
+            SERVICES+=('-s')
+            ;;
+        -h | --help | help)
+            start_environment__usage
+            exit 0
+            ;;
+        *)
+            if [ -n "${command}" ]; then
+                echo "${red}Unknown option:${reset} ${command}"
+            else
+                start_environment__usage
+                exit 1
+            fi
+            ;;
+        esac
+    done
+
+    banner
+
+    # Import environment
+    set -a
+    . ./.env
+    set +a
+
+    echo "░░░ Starting PlaceOS <${PLACEOS_TAG}>"
+    echo "░░░ For detailed logging, run \`tail -f ${logfile}\`"
+
+    PLACE_DOMAIN=${PLACE_DOMAIN:-"localhost:8443"}
+    PLACE_EMAIL=${PLACE_EMAIL:-"support@place.tech"}
```

Comment:
Could we try a pull the current users email from their git config, then (silently) fall back to this?

Reply:
Nice idea

Reply:
Could be part of a wizard if a `.env.email` is not present, or an explicit email arg present.